server/upsd.c

```diff
@@ -1270,7 +1270,16 @@ int main(int argc, char **argv)
 	poll_reload();
 
 	if (num_ups == 0) {
-		fatalx(EXIT_FAILURE, "Fatal error: at least one UPS must be defined in ups.conf");
+		char *envvar = getenv("NUT_NOCONF_ALLOWED");
+		if ( (envvar != NULL) && (0 == strncasecmp("TRUE", envvar, 4)) ) {
```

Comment:
`strncasecmp("TRUE", ..., 4)` is redundant, consider using `strcasecmp("TRUE", ...)` in this case.

In general, avoid usage of `strn*()` functions if possible. They are poorly suited for C-style strings and are an endless source of bugs and security issues.

Reply:
I was told the opposite, that the routines without "n" with random-memory pointers are an endless source of bugs and stack overflows... whom to trust? :p

Reply:
Delegating to @aquette - I think majority of NUT codebase uses `strn<something>` so this would be at least consistent.

Reply:
I looked at OpenBSD's take on the string functions for guidance and it looks like only the functions with dangerous write buffer semantics are actually banned, so using `strnlen()/strncmp()/strncasecmp()` is fine for them. Furthermore, they do use `strcasecmp()` when there's a constant string literal for one of the parameters. So I've been mistaken on my crusade against _all_ `strn*()` functions, but the change to `strcasecmp()` here is still warranted in this case.